### ISSUE TITLE
HAN - Save, Load and Test

### ISSUE THREAD

**Issue by truas** [Hsankesara/DeepResearch#3]
Hello,

Is it possible to include some instructions/template for saving the HAN model, and loading them to re-use?

In the readme we already have the instruction to run the HAN:
`han_network = HAN.HAN(text = df.text, labels = df.category, num_categories = total_categories, pretrained_embedded_vector_path = embedded_vector_path, max_features = max_num_of_features, max_senten_len = max_sentence_len, max_senten_num = max_sentence_num , embedding_size = size_of_embedded_vectors)`

If you include a couple of instructions suggesting how to save (I believe the version for the notebook already has), load and use the model that would be great. 

Regards,
T

**Comment by Hsankesara** [comment]
Hey @truas,
There is a feature to save the HAN model:
`train_model(self, epochs, batch_size, best_model_path = None, final_model_path = None, plot_learning_curve = True)`
Here you can save both the best model (Minimum Validation loss) and final_model. Just give it path where you want to save it. For Example:
`han_network.train_model(epochs=3, batch_size=16,
                            best_model_path='./best_model.h5')`
To know more about save and load model go [here](https://machinelearningmastery.com/save-load-keras-deep-learning-models/)

Yeah, I haven't included file loading the saved weights. I added it to my todo list and I'll work on it as soon as possible. :slightly_smiling_face:

**Comment by truas** [comment]
@Hsankesara ,

Thank you !

No worries, you contribution is a great one already. These "new" embedding models (e.g. ELMo, HAN, USE, Infesent) seem to be a great thing, but none of them are as easy to work as word2vec/doc2vec.

Keep it up the great work !

Regards,
T.

**Comment by Hsankesara** [comment]
Thanks @truas for your time in this repository.
